javaparser-core-testing/src/test/resources/com/github/javaparser/EscapeSequences.java

```diff
@@ -0,0 +1,32 @@
+package com.github.javaparser;
+
+public class EscapeSequences {
+
+    public static void main(String[] args) {
+        Object[] chars = {
+                '\\', '\u005C\u005C', '\u005c\u005c',
+                "---",
+                '\n', '\u005cn', '\u005Cn',
+                "---",
+                '\r', '\u005cr', '\u005Cr',
+                "---",
+                '\t', '\u005ct', '\u005Ct',
+                "---",
+                '\b', '\u005cb', '\u005Cb',
+                "---",
+                '\f', '\u005cf', '\u005Cf',
+                "---",
+                '\'', '\u005c'', // '\u005C'', // that's weird, this last one won't compile
```

Comment:
`\u005C` is a back-slash character

This means that `'\u005C'` becomes `'\'` which, reasonably, does not compile.

If you change it to `\u005C\u005C` then this does compile (equivalent to `'\\'`).

Reply:
I agree that `'\u005C'` becomes `'\'` and that would be invalid, however the sequence that i'm escaping has a double single quote.

Just like the lowercase version `'\u005c''` gets interpreted as `\'` I would have expected the upper case version to do the same but something about this sequence won't compile.

`\u005C\u005C` is already covered a few lines above

Reply:
Actually, reading more into it, I'm starting to realise why this is odd -- the Unicode escape sequences should be case insensitive (as demonstrated in the other examples) so it's strange that this specific example doesn't compile.